public/app/plugins/datasource/cloud-monitoring/datasource.ts

```diff
@@ -137,7 +137,8 @@ export default class CloudMonitoringDatasource extends DataSourceApi<CloudMonito
     const queries = options.targets
       .map(this.migrateQuery)
       .filter(this.shouldRunQuery)
-      .map(q => this.prepareTimeSeriesQuery(q, options));
+      .map(q => this.prepareTimeSeriesQuery(q, options.scopedVars))
+      .map(q => ({ ...q, intervalMs: options.intervalMs, type: 'timeSeriesQuery' }));
```

Comment:
Hmm this one new line seems like it is fixing something else than the templating isn't it? If it actually is related why is this needed?

Reply:
actually I just moved it from `prepareTimeSeriesQuery` in https://github.com/grafana/grafana/pull/27354/files#diff-3c4534844b9cdbcc38d12b8239b7dfddL344 so that `prepareTimeSeriesQuery` returns a `CloudMonitoringQuery` instead of `DataQueryRequest<CloudMonitoringQuery>` which is then ugmented here. That way I can re-use it in `interpolateVariablesInQueries` without having issues with types.

Reply:
Ah ok did not notice that, good idea.